File: app/src/release/java/theo/base/dagger/BuildVariantModule.java
package theo.base.dagger;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import theo.base.AppInitializer;
import theo.base.ReleaseAppInitializer;

@Module
public class BuildVariantModule {

    @Singleton
    @Provides
    AppInitializer provideReleaseAppInitializer(){
        return new ReleaseAppInitializer();
    }
}

File: app/src/release/java/theo/base/dagger/BuildVariantModule.kt
package theo.base.dagger

import javax.inject.Singleton

import dagger.Module
import dagger.Provides
import theo.base.AppInitializer
import theo.base.ReleaseAppInitializer

@Module
class BuildVariantModule {

    @Singleton
    @Provides
    internal fun provideReleaseAppInitializer(): AppInitializer {
        return ReleaseAppInitializer()
    }
}
